### PR TITLE
Make kmsKey optional

### PR DESCRIPTION
**Pull request by winjer**
you don't need to specify the key, since it is defined in the sops encrypted file.

this change makes the key optional.

### CODE REVIEW

File: provider/main.py [hunk 1]
```diff
@@ -4,10 +4,14 @@
...
 import logging
 
-def sops_decode(data, kms_key, data_format):
+def sops_decode(data, data_format, kms_key=None):
     dir_path = os.path.dirname(os.path.realpath(__file__))
     sops_binary = os.path.join(dir_path, 'sops')
-    output = subprocess.run([sops_binary, '--kms', kms_key, '-d', '--input-type', data_format, '--output-type', 'json', '/dev/stdin'], input=data, capture_output=True)
+    command = [sops_binary, '-d', '--input-type', data_format, '--output-type', 'json']
+    if kms_key:
+        command.extend(['--kms', kms_key, ])
```
Comment (plumdog) [inline provider/main.py]: Superfluous trailing `, `

File: provider/main.py [hunk 2]
```diff
@@ -74,7 +78,7 @@ def on_create(event):
         data_type = fileType
         if not data_type:
             data_type = s3Path.rsplit('.', 1)[-1]
-        secrets = sops_decode(raw_content, kmsKey, data_type)
+        secrets = sops_decode(raw_content, data_type, kmsKey)
```
Comment (plumdog) [inline provider/main.py]: Oh god I've just realised what a horrible job of the camel-vs-snake casing I've done in this file who even am I?

File: provider/main.py [hunk 3]
```diff
@@ -56,7 +60,7 @@ def get_mapped_values(secrets, mappings):
 def on_create(event):
     logging.info('On create')
 
-    kmsKey = event['ResourceProperties']['KMSKeyArn']
+    kmsKey = event['ResourceProperties'].get('KMSKeyArn', None)
```
Comment (plumdog) [inline provider/main.py]: Don't need the `None`

Comment (winjer) [reply provider/main.py]: TIL

File: provider/main.py (hunk 2 above)
Comment (winjer) [reply provider/main.py]: ikr

**Comment by plumdog** [comment]
@winjer Merged and released as 1.3.0 -  https://www.npmjs.com/package/sops-secretsmanager-cdk/v/1.3.0

**Pull request merged**